src/ray/gcs/redis_gcs_client.h

```diff
@@ -18,7 +18,7 @@ namespace gcs {
 class RedisContext;
 
 class RAY_EXPORT RedisGcsClient : public GcsClientInterface {
-  friend class ActorStateAccessor;
+  friend class RedisActorStateAccessor;
```

Comment:
Not introduced by your PR, but this is a good time to rethink interfaces. Can we please come up with a solution that doesn't require `friend`ing this class? This is a red flag for having leaky abstractions.

Reply:
Will remove those friend class after we replace `RedisGcsClient` with interface class `GcsClient`.

Now, we make `xxx_table()` as protected functions to prevent developer  incorrect use of old interfaces. At the same time those new implementation `RedisXXXAccessor` need to use those functions, so we make it as friend class.